app/destination/page.js

```diff
@@ -1,114 +1,68 @@
-      <b>{name.toUpperCase()}</b>
-      <button onClick={onRemove}>remove</button>
-    </div>
-  );
-}
+import PlanetWishlistItem from '@/components/destination/PlanetWishlistItem';
 
 
 export const Destinations = () => {
-  const [selectedPlanets, onAddPlanet] = useState([]);
-
-  let isPlanetSelected = false;
-  let numberOfPlanets = 0;
+  const [selectedPlanets, setSelectedPlanets] = useState([]);
 
-  const onAddOrRemovePlanet = (name, index) => {
-    // TASK - React 1 week 2
-    // Implement this function
-    // If you press the "ADD PLANET" the selected planet should display "SELECTED"
-    // And the counter should update, how many planets are selected (numberOfPlanets)
-    console.log(`You seleceted the following planet: ${name}, with the index of ${index}`);
-  }
+  const onAddOrRemovePlanet = (name, thumbnail) => {
+    if (selectedPlanets.some(p => p.name === name)) {
+      setSelectedPlanets(prev => prev.filter(planet => planet.name !== name));
+    } else {
+      setSelectedPlanets(prev => [...prev, { name, thumbnail }]);
+    }
+  };
 
   return (
     <div className="fullBGpicture">
       <main className="mainContent">
         <h1>Travel destinations</h1>
+
         <section className="card">
           <h2>Wishlist</h2>
-          {/* TASK - React 1 week 2 */}
-          {/* Display the number Of selected planets */}
-          {/* Display the "no planets" message if it is empty! */}
-          <p>No planets in wishlist :(</p>
-          <p>You have {numberOfPlanets} in your wishlist</p>
-          <b>List coming soon after lesson 3!</b>
-          
-          {/* STOP! - this is for week 3!*/}
-          {/* TASK - React 1 week 3 */}
-          {/* Import the AddWishlistItem react component */}
-          {/* <AddWishlistItem /> */}
-          {/* TASK - React 1 week 3 */}
-          {/* Convert the list, so it is using selectedPlanets.map() to display the items  */}
-          {/* Implement the "REMOVE" function */}
-          {/* uncomment the following code snippet: */}
-          {/* 
-          <h3>Your current wishlist</h3>
+          <p>{selectedPlanets.length === 0 ? "No planets in wishlist :(" : `You have ${selectedPlanets.length} in your wishlist`}</p>
+
           <div className={styles.wishlistList}>
-            <PlanetWishlistItem 
-              name="europa"
-              onRemove={() => removeFromWishlist('europa')}
-              thumbnail="/destination/image-europa.png"
-            />
-            <PlanetWishlistItem 
-              name="europa"
-              onRemove={() => removeFromWishlist('europa')}
-              thumbnail="/destination/image-europa.png"
-            />
-          </div> */}
+            {selectedPlanets.map((planet, index) => (
+              <PlanetWishlistItem
+                key={index}
+                name={planet.name}
+                onRemove={() => setSelectedPlanets(prev => prev.filter(p => p.name !== planet.name))}
```

Comment:
That's good, but maybe not super readable.  I suggest putting the logic of onRemove inside a function and then pass it to the onRemove to make things more clean